app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,36 @@
+class VotesController < ApplicationController
```

Comment:
having a `VotesController` with `upvote` and `downvote` actions totally works. However if we want to follow more of the rails way it would be more conventional to have an `UpVotesController` and `DownVotesController` each with a `create` action.

Refactoring this would involve simply changing the urls that your AJAX methods post to and moving this controller code into two separate controller files.